tst/utils/Thread.cpp

```diff
@@ -174,16 +180,29 @@ TEST_F(ThreadFunctionalityTest, NegativeTest)
 {
     TID threadId = 0;
     SIZE_T threadStack = 512 * 1024; // 0.5 MiB
+    ThreadParams threadParams;
+    threadParams.version = 0;
+    threadParams.stackSize = threadStack;
 
     // No out value case
-    EXPECT_NE(STATUS_SUCCESS, THREAD_CREATE_WITH_PARAMS(NULL, emptyRoutine, threadStack, NULL));
+    EXPECT_NE(STATUS_SUCCESS, THREAD_CREATE_WITH_PARAMS(NULL, &threadParams, emptyRoutine, NULL));
 
     // Request too large stack size case
-    EXPECT_NE(STATUS_SUCCESS, THREAD_CREATE_WITH_PARAMS(&threadId, emptyRoutine, SIZE_MAX, NULL));
+    threadParams.stackSize = SIZE_MAX;
+    EXPECT_NE(STATUS_SUCCESS, THREAD_CREATE_WITH_PARAMS(&threadId, &threadParams, emptyRoutine, NULL));
     EXPECT_EQ(0, threadId);
 
     // No out value case
-    EXPECT_NE(STATUS_SUCCESS, THREAD_CREATE(NULL, testThreadRoutine, NULL));
+    EXPECT_NE(STATUS_SUCCESS, THREAD_CREATE(NULL, emptyRoutine, NULL));
+
+    // Invalid version
+    threadParams.version = THREAD_PARAMS_CURRENT_VERSION + 100;
+    EXPECT_NE(STATUS_SUCCESS, THREAD_CREATE_WITH_PARAMS(&threadId, &threadParams, emptyRoutine, NULL));
```

Comment:
For as many of these failing test cases as possible, rather than checking for !SUCCESS, please check for the expected error code (so will be using EXPECT_EQ rather than EXPECT_NE).

Reply:
I made it NE SUCCESS for consistency. I can adjust it for this file for this PR. We can look into adjusting this for the rest of the codebase in the future as this should help prevent regressions.